apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java

```diff
@@ -175,4 +182,67 @@ static ByteString httpRequestBody(Operation operation, ScalarTypeAdapters scalar
     jsonWriter.close();
     return buffer.readByteString();
   }
+
+  static RequestBody transformToMultiPartIfUploadExists(RequestBody originalBody, Operation operation)
+      throws IOException {
+    //ArrayList<GraphqlUpload> allUploads = new ArrayList<>();
+    HashMap<String, String[]> filesMap = new HashMap<>();
```

Comment:
Not sure why second type is an array `HashMap<String, String[]>`?
I think that you can instead of 2 maps have this structure: `List<FileUpload>` where:
```
FileUpload(key:String, mimeType:String, file:File)
```

`key`  = `"variables." + variableName`
`mimeType` = `GraphqlUpload.mimeType`
`file` = `GraphqlUpload.file`

The order in the list is the index in multipart body.
What do you think?

Reply:
You're right. I will revise the code.